src/i_video.c

```diff
@@ -1754,38 +1694,38 @@ static void CreateSurfaces(int w, int h)
 
     if (screenbuffer != NULL)
     {
-        SDL_FreeSurface(screenbuffer);
+        SDL_DestroySurface(screenbuffer);
     }
 
-    screenbuffer = SDL_CreateRGBSurface(0, w, h, 8, 0, 0, 0, 0);
-    SDL_FillRect(screenbuffer, NULL, 0);
+    screenbuffer = SDL_CreateSurface(w, h, SDL_PIXELFORMAT_INDEX8);
+
+    I_SetPalette(W_CacheLumpName("PLAYPAL", PU_CACHE));
+    SDL_SetSurfacePalette(screenbuffer, palette);
 
     I_VideoBuffer = screenbuffer->pixels;
     V_RestoreBuffer();
 
     if (argbbuffer != NULL)
     {
-        SDL_FreeSurface(argbbuffer);
+        SDL_DestroySurface(argbbuffer);
     }
 
     // [FG] create intermediate ARGB frame buffer
 
-    argbbuffer = SDL_CreateRGBSurfaceWithFormatFrom(
-        NULL, w, h, 0, 0, SDL_PIXELFORMAT_ARGB8888);
-
-    I_SetPalette(W_CacheLumpName("PLAYPAL", PU_CACHE));
-
-    // [FG] create texture
+    argbbuffer = SDL_CreateSurfaceFrom(w, h, SDL_GetWindowPixelFormat(screen),
+                                       NULL, w * 4);
 
     if (texture != NULL)
     {
         SDL_DestroyTexture(texture);
     }
 
-    texture = SDL_CreateTexture(renderer, SDL_PIXELFORMAT_ARGB8888,
```

Comment:
Could this change be a cause for the performance regressions we see?

Reply:
We don't have any performance regressions, SDL2 direct3d11 also works badly on MrAlaux's machine. The problem is that SDL3 now chooses direct3d11 as default backend. It works faster on my machine though.